taf/api/roles.py

```diff
@@ -908,3 +908,18 @@ def _update_role(
         auth_repo.update_role_keystores(role, keystore_keys, write=False)
     if len(yubikeys):
         auth_repo.update_role_yubikeys(role, yubikeys, write=False)
+
+
+def list_roles(auth_repo: AuthenticationRepository) -> None:
+    """
+    Print a list of all defined roles with their thresholds and parent roles.
+    """
+    all_roles = auth_repo.get_all_roles()
+    for role in all_roles:
+        threshold = auth_repo.get_role_threshold(role)
+        if role not in MAIN_ROLES:
+            parent_role = auth_repo.find_delegated_roles_parent(role)
```

Comment:
Could you try to print this so that each delegated target roles is printed underneath its parent role? So,
```
target (threshold: 2)
  law (threshold: 1)
  docs (threshold 1)
...
timestamp
```
A simple recursive function which prints the parent and calls print again to print children should be sufficient, we will never have so many roles that such a recursion is too slow.